### PR TITLE
Replace HTTP client with Akka HTTP.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This is an example of how the universe package JSON could be loaded with
Akka Http and processed asynchrnously with Akka Streams.

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Retry.scala [hunk 1]
```diff
@@ -0,0 +1,191 @@
+package com.mesosphere.cosmos
```
Comment (jeschkies) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Retry.scala:1]: This has to come from USI.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/repository/UniverseClientSpec.scala [hunk 2]
```diff
@@ -109,34 +117,30 @@ final class UniverseClientSpec extends FreeSpec with Matchers {
...
 
-    "should retry before failing to fetch a bad host" in {
-      val version = universe.v3.model.DcosReleaseVersionParser.parseUnsafe("0.0")
-      val repoUri = "https://something-that-is-never.valid" / "doesnotexist.json"
-      val expectedPkgRepo = PackageRepository("badRepo", repoUri)
-      val result = universeClient(expectedPkgRepo, version).liftToTry
-      assertThrows[com.twitter.util.TimeoutException](
-        Await.result(
-          result,
+//    "should retry before failing to fetch a bad host" in {
```
Comment (timcharper) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/repository/UniverseClientSpec.scala:127]: Why is this commented out?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala [hunk 3]
```diff
@@ -1,178 +1,112 @@
...
-        sr.scope("status").counter(status.toString).incr()
-        // Different forms of redirect, HttpURLConnection won't follow a redirect across schemes
-
-        val loc = Option(conn.getHeaderField("Location")).map(Uri.parse).flatMap(_.schemeOption)
-        throw UnsupportedRedirect(List(uri.schemeOption.get), loc).exception
-      case status =>
-        sr.scope("status").counter(status.toString).incr()
-        throw GenericHttpError(uri = uri, clientStatus = HttpResponseStatus.valueOf(status)).exception
+  private[this] def performRequest(request: HttpRequest)(implicit ec: ExecutionContext, system: ActorSystem): Future[HttpResponse] = async {
+    val response = await(Http().singleRequest(request))
```
Comment (timcharper) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala:83]: Does Cosmos make a lot of http requests to the same host? May be worth using a connection pool?

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/repository/UniverseClientSpec.scala (hunk 2 above)
Comment (jeschkies) [reply cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/repository/UniverseClientSpec.scala:127]: Good catch. I had no retrying implemented yet.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala (hunk 3 above)
Comment (jeschkies) [reply cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala:83]: It does make them to the universe. We have https://github.com/dcos/cosmos/blob/master/cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/PackageCollection.scala#L24 to reduce the number.

**Comment by jeschkies** [comment]
Won't do.

**Pull request closed without merging**